plugins/templates/removeTag.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+
+/* Currently, only build scope is allowed to tag template due to security reasons.
+ * The same pipeline that publishes the template has the permission to tag it.
+ */
+module.exports = () => ({
+    method: 'DELETE',
+    path: '/templates/tags',
+    config: {
+        description: 'Delete a template tag',
+        notes: 'Delete a specific template',
+        tags: ['api', 'templates'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['build']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const templateFactory = request.server.app.templateFactory;
+            const templateTagFactory = request.server.app.templateTagFactory;
+            const pipelineId = request.auth.credentials.pipelineId;
+            const config = request.payload;
```

Comment:
I think it is unusual, but not forbidden according to [RFC 7231]( https://tools.ietf.org/html/rfc7231#section-4.3.5). 
A couple posts related to this:

- https://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request
- https://github.com/zalando-stups/lizzy/issues/151


I would say it's weird and we should avoid it if we can. But I think in this case, it makes sense to send a payload with it. An alternative would be delete `templates/tags/{name}/{tag}`?  I'm not sure if that's weird either. What do you think?